src/core/DY.Core.Trace.Base.fst

```diff
@@ -204,6 +204,22 @@ let prefix_grows #label_t tr i =
   reveal_opaque (`%grows) (grows #label_t);
   norm_spec [zeta; delta_only [`%prefix]] (prefix #label_t)
 
+
+/// The relation <$ has Nil as a least element, and
+/// is compatible with growing traces via Snoc
+
+let grows_nil (#label_t:Type) (tr:trace_ label_t)
+  : Lemma (ensures Nil <$ tr)
+    [SMTPat (Nil <$ tr)]
+  = reveal_opaque (`%grows) (grows #label_t)
+
+let grows_snoc (#label_t:Type) (tr:trace_ label_t) (e:trace_entry_ label_t)
+  : Lemma (ensures tr <$ (Snoc tr e))
+    [SMTPat (tr <$ (Snoc tr e))]
+  = reveal_opaque (`%grows) (grows #label_t);
+    norm_spec [zeta; delta_only [`%prefix]] (prefix #label_t)
```

Comment:
In general I think that since we now have `append_entry` we should try to use it as much as possible (to avoid having two names for the same thing). It's fine to use `Snoc` when necessary but in that case it's useful to have a comment explaining the technical reasons for this choice.

Reply:
Yes, I would generally agree --- do you think the comment above, at the start of this block of functions is sufficient, or would you rather have it directly before each function where this happens individually?

Reply:
(or alternatively, do you see a way to make the lemmas trigger with the same robustness as using Nil and Snoc directly, while only using `empty_trace` and `append_entry`?)

Reply:
I think the comment is good.